### PR TITLE
Pull request for #97: Anything builder

### PR DESCRIPTION
**Pull request by stanfordquan**
Please see #97 for discussion. There's a branch on the template that will demonstrate a use case. I will update the template FAQ after this pull request and before the issue closes.

### CODE REVIEW

**Comment by stanfordquan** [comment]
@Shun-Yang these two links may help. I borrowed the structure from [`env.Command`](http://scons.org/doc/HTML/scons-api/SCons.Environment-pysrc.html#Base.Command) and the [`Builder`](http://scons.org/doc/HTML/scons-api/SCons.Builder-module.html) module help page explains a bit more about what a Builder/Node is.

File: gslab_scons/builders/build_anything.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        target_dir  = misc.get_directory(target_file)
+        try:
+            log_ext = '_%s' % env['log_ext']
+        except KeyError:
+            log_ext = ''
+        log_file = os.path.join(target_dir, ('sconscript%s.log' % log_ext))
+
+        # System call
+        try:
+            command = '%s > %s' % (action, log_file)
```
Comment (Shun-Yang) [inline gslab_scons/builders/build_anything.py]: What about `command = '%s > %s  2>&1' % (action, log_file)` to also log `stderr`?

File: gslab_scons/builders/build_anything.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        log_ext: string
+            Instead of logging to `sconscript.log` in the target dir, 
+            the builder will log to `sconscript_<log_ext>.log`.
+        origin_log_file: string
+            Sometimes, your command may produce a log file in an undesirable location.
+            Specifying the that location in this argument leads the builder to append
+            the content of origin_log_file to log_file and delete origin_log_file.            
+            The builder will crash if this file doesn't exist at the end of the command.
+        warning: Boolean
+            Turns off warnings if warning == False.
```
Comment (Shun-Yang) [inline gslab_scons/builders/build_anything.py]: `warning = False`?

**Comment by stanfordquan** [comment]
1) Done, thanks!

2) Added newline.

3) Yes we can make it say `build_anything`, but then the action generator needs a different name. How about `_build_anything` for the generator, and then `build_anything` for the main event?

4) Basically, no. `Stata` has this funky thing of producing a log at the root directory with the name of a dofile, I was trying to prevent that sort of behavior (which is, so far, only necessary in the Stata case). In the other builders you almost always catch all your logs with the redirection operator `>`.  

5) Agreed.

**Comment by stanfordquan** [comment]
@Shun-Yang I made another change in the `_build_anything` `warning` syntax. Basically because it's dealt with within the generator and not the builder, it's passed as a keyword rather than an `env` element.

I made the warning slightly prettier.

image

**Comment by Shun-Yang** [comment]
@stanfordquan , I think the directory `anything_test` is accidentally merged to template 4.1.0 branch. Should we delete it?

**Comment by stanfordquan** [comment]
Yes @Shun-Yang . I think I was on the wrong branch when I made that commit.

**Comment by stanfordquan** [comment]
Done as of7dfbad4.

**Pull request merged**